src/Symfony/Component/Messenger/Transport/Sender/SendersLocator.php

```diff
@@ -41,20 +42,33 @@ public function __construct(array $sendersMap, ContainerInterface $sendersLocato
      */
     public function getSenders(Envelope $envelope): iterable
     {
-        $seen = [];
+        if (\count($envelope->all(OverrideSendersStamp::class)) > 0) {
+            $overrideSendersStamp = $envelope->last(OverrideSendersStamp::class);
 
-        foreach (HandlersLocator::listTypes($envelope) as $type) {
-            foreach ($this->sendersMap[$type] ?? [] as $senderAlias) {
-                if (!\in_array($senderAlias, $seen, true)) {
-                    if (!$this->sendersLocator->has($senderAlias)) {
-                        throw new RuntimeException(sprintf('Invalid senders configuration: sender "%s" is not in the senders locator.', $senderAlias));
-                    }
+            foreach ($overrideSendersStamp->getSenders() as $senderAlias) {
+                yield from $this->getSenderFromAlias($senderAlias);
+            }
+        } else {
```

Comment:
You could use early return here and remove this `else` statement. That would simplify the diff